lib/passport/passport.strategy.ts

```diff
@@ -1,11 +1,12 @@
 import * as passport from 'passport';
 import { Type } from '../interfaces';
+import { Value } from '../types';
 
 export function PassportStrategy<T extends Type<any> = any>(
   Strategy: T,
   name?: string | undefined
 ): {
-  new (...args): T;
+  new (...args): Value<T>;
```

Comment:
Shouldn't we change this line:
```
Strategy: T,
```
instead?

Reply:
You mean `Strategy: Type<T>` and `T extends any`? That was my first try and it did not work.